internal/templates/render/render.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package render
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/abcxyz/pkg/cli"
+)
+
+type Command struct {
+	cli.BaseCommand
+
+	template        string
+	spec            string
+	dest            string
+	gitProtocol     string
+	logLevel        string
+	allowNonGitDest bool
+	forceOverwrite  bool
+	keepTempDirs    bool
+	inputs          map[string]string
+}
+
+// Desc implements cli.Command.
+func (c *Command) Desc() string {
+	return "Instantiate a template to setup a new app or add config files"
+}
+
+func (c *Command) Help() string {
+	return `
+Usage: {{ COMMAND }} [options]
+
+  The {{ COMMAND }} command renders the given template into the given dest
+  directory.
+`
+}
+
+func (c *Command) Flags() *cli.FlagSet {
+	set := cli.NewFlagSet()
+
+	sect := set.NewSection("Render options")
+	sect.StringVar(&cli.StringVar{
+		Name:    "template",
+		Aliases: []string{"t"},
+		Example: "helloworld@v1",
+		Target:  &c.template,
+		Usage: `the location of the template to be instantiated. Many forms are accepted. ` +
+			`"helloworld@v1" means "github.com/abcxyz/helloworld repo at revision v1; this is for a template owned by abcxyz. ` +
+			`"myorg/helloworld@v1" means github.com/myorg/helloworld repo at revision v1; this is for a template not owned by abcxyz but still on GitHub. ` +
+			`"mygithost.com/mygitrepo/helloworld@v1" is for a template in a remote git repo but not owned by abcxyz and not on GitHub. ` +
+			`"mylocaltarball.tgz" is for a template not in git but present on the local filesystem. ` +
+			`"http://example.com/myremotetarball.tgz" os for a non-Git template in a remote tarball.`,
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "spec",
+		Aliases: []string{"s"},
+		Example: "path/to/spec.yaml",
+		Target:  &c.spec,
+		Default: "./spec.yaml",
+		Usage:   "the path of the .yaml file within the template directory that specifies how the template is rendered.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "dest",
+		Aliases: []string{"d"},
+		Example: "/my/git/dir",
+		Target:  &c.dest,
+		Default: ".",
+		Usage:   "the target directory in which to write the output files.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "git-protocol",
```

Comment:
I thought we agreed to leave this to user input? Like I can specify ssh in my `-t` (or `-n` as I proposed) flag.

Reply:
The implementation here is what is in the design doc. See https://docs.google.com/document/d/1Ej5DprrjHGtX9oRl2QARQ0AXfw9qUvxRFFOaq2tqa2M/edit#bookmark=id.5s95rmhwtpz7.

Reply:
My memory could be failing me. I recall some discussion in the comments but maybe it was from an older iteration. cc @sethvargo

Reply:
You *can* specify SSH or HTTPS, but when the input is ambiguous, we will apply the environment/flag config.